common4j/src/main/com/microsoft/identity/common/java/authorities/Authority.java

```diff
@@ -157,9 +168,11 @@ public static Authority getAuthorityFromAuthorityUrl(String authorityUrl) {
             } else if (CIAM.equalsIgnoreCase(authorityTypeStr)) {
                 authority = new CIAMAuthority(authorityUrl);
             } else if (AAD_NA.equalsIgnoreCase(authorityTypeStr) && configuredAuthority instanceof NativeAuthCIAMAuthority) {
-                authority = new NativeAuthCIAMAuthority(
-                        authorityUrl,
-                        ((NativeAuthCIAMAuthority) configuredAuthority).getClientId());
+                if (clientId == null) {
+                    authority = new CIAMAuthority(authorityUrl);
```

Comment:
Can you add a comment explaining why it's not a problem in our flows that a CIAMAuthority is created here?

Reply:
Done